src/script/interpreter.cpp

```diff
@@ -616,7 +616,36 @@ bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript&
                     break;
                 }
 
-                case OP_NOP1: case OP_NOP4: case OP_NOP5:
+                case OP_CHECKTEMPLATEVERIFY:
+                {
+                    // if flags not enabled; treat as a NOP4
+                    if (!(flags & SCRIPT_VERIFY_DEFAULT_CHECK_TEMPLATE_VERIFY_HASH)) {
+                        if (flags & SCRIPT_VERIFY_DISCOURAGE_UPGRADABLE_NOPS)
```

Comment:
Oh, I think I see what's causing your error in #22865. You don't need to fail on `SCRIPT_VERIFY_DISCOURAGE_UPGRADABLE_NOPS` now that you've repurposed this NOP.

Reply:
This is actually a new line I added -- it's required for correctness if this is to be merged/released before setting release params. https://github.com/bitcoin/bitcoin/pull/21702#discussion_r701305405

However, it is not the reason there is a problem. It is also a problem later on where !=32 byte arguments are upgradable too.

Reply:
resolving as per https://github.com/bitcoin/bitcoin/issues/22865#issuecomment-914679373.

this approach, although mildly (temporarily) problematic, yields safer results until CTV is active.